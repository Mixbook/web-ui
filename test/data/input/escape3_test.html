<!DOCTYPE html>
<!--
Copyright (c) 2012, the Dart project authors.  Please see the AUTHORS file
for details. All rights reserved. Use of this source code is governed by a
BSD-style license that can be found in the LICENSE file.
-->
<html lang="en">
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">
  <script src="packages/web_ui/testing/testing.js"></script>
</head>
<body>
  <!-- Tests it is valid to use existing nodes as SafeHtml. -->
  <div>text to text (expect &lt;span&gt;text1&lt;/span&gt;): {{w}}</div>
  <div>text to html (expect html1 in a real span): {{x}}</div>
  <div>html to text (expect &lt;span&gt;text2&lt;/span&gt;): {{y}}</div>
  <div>html to html (expect html2 in a real span): {{z}}</div>
  <script type="application/dart">
  import 'dart:async';
  import 'dart:html';
  import 'package:unittest/unittest.dart';
  import 'package:web_ui/observe.dart';
  import 'package:web_ui/safe_html.dart';

  @observable var w = '<span>text0</span>';
  @observable var x = '<span>text0</span>';
  @observable var y = new Element.html('<span>html0</span>');
  @observable var z = new Element.html('<span>html0</span>');
  main() {
    Timer.run(() {
      w = '<span>text1</span>';
      x = new Element.html('<div>html1</div>');
      y = '<span>text2</span>';
      z = new Element.html('<div>html2</div>');
      window.postMessage('done', '*');
    });
  }
  </script>
</body>
</html>
