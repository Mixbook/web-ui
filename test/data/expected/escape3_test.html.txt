Content-Type: text/plain
layer at (0,0) size 800x600
  RenderView at (0,0) size 800x600
layer at (0,0) size 800x124
  RenderBlock {HTML} at (0,0) size 800x124
    RenderBody {BODY} at (8,8) size 784x108
      RenderBlock {DIV} at (0,0) size 784x18
        RenderText {#text} at (0,0) size 420x18
          text run at (0,0) width 420: "text to text (expect <span>text1</span>): "
        RenderText {#text} at (420,0) size 180x18
          text run at (420,0) width 180: "<span>text1</span>"
      RenderBlock {DIV} at (0,18) size 784x36
        RenderBlock (anonymous) at (0,0) size 784x18
          RenderText {#text} at (0,0) size 430x18
            text run at (0,0) width 430: "text to html (expect html1 in a real span):"
        RenderBlock {DIV} at (0,18) size 784x18
          RenderText {#text} at (0,0) size 50x18
            text run at (0,0) width 50: "html1"
      RenderBlock {DIV} at (0,54) size 784x18
        RenderText {#text} at (0,0) size 420x18
          text run at (0,0) width 420: "html to text (expect <span>text2</span>): "
        RenderText {#text} at (420,0) size 180x18
          text run at (420,0) width 180: "<span>text2</span>"
      RenderBlock {DIV} at (0,72) size 784x36
        RenderBlock (anonymous) at (0,0) size 784x18
          RenderText {#text} at (0,0) size 430x18
            text run at (0,0) width 430: "html to html (expect html2 in a real span):"
        RenderBlock {DIV} at (0,18) size 784x18
          RenderText {#text} at (0,0) size 50x18
            text run at (0,0) width 50: "html2"
